### ISSUE TITLE
Classes in Java sources are not available during linting

### ISSUE THREAD

**Issue by emlyn** [borkdude/clj-kondo#877]
**version**

clj-kondo v2020.05.02

**platform**

JVM

**problem**

I have a class defined in a Java source file. Referencing that class generates a false positive unresolved namespace warning.

**repro**

 I've put together a small project with a minimal reproduction [here](https://github.com/emlyn/kondo-test) (basically a minimal project with one Java class and one Clojure namespace that references the Java class).

**expected behavior**

The reference to the Java class should not result in a warning.
Classes defined in a Java source file should be on the classpath during linting

Is this a bug? Or is there something else I should be doing to make it recognise the Java class?

**Comment by borkdude** [comment]
Hi @emlyn. It's pretty standard for Java packages to not use a single segment package name. Clj-kondo uses this to differentiate between Java classes and Clojure vars.

There is no true way to tell if this is a Java class or call to a Clojure var:

```
kondo_test.TestClass/foo
```

Clj-kondo uses this function to guess the difference:

```
(defn class-name? [s]
  (let [splits (str/split s #"\.")]
    (and (> (count splits) 2)
         (Character/isUpperCase ^char (first (last splits))))))
```

So if you add one more level to your package name: `foo.bar.TestClass` then it will work as expected.

**Comment by emlyn** [comment]
Thank you for explaining, that makes sense. I've added another segment to the package name and it works now :-)
